router/core/header_rule_engine_test.go

```diff
@@ -54,6 +54,42 @@ func TestPropagateHeaderRule(t *testing.T) {
 
 	})
 
+	t.Run("Should propagate repeated header names", func(t *testing.T) {
+
+		ht, err := NewHeaderPropagation(&config.HeaderRules{
+			All: &config.GlobalHeaderRule{
+				Request: []*config.RequestHeaderRule{
+					{
+						Operation: "propagate",
+						Named:     "X-Test-1",
+					},
+				},
+			},
+		})
+		assert.Nil(t, err)
+
+		rr := httptest.NewRecorder()
+
+		clientReq, err := http.NewRequest("POST", "http://localhost", nil)
+		require.NoError(t, err)
+		clientReq.Header.Add("X-Test-1", "test1")
```

Comment:
manually modify the header here with not canonical names

Reply:
I don't think this is where it belongs. It doesn't seem like it's the rules engines job to canonicalize the incoming headers. At least, it's not doing it before this change. This change isn't making any change to any of the code that's handling incoming headers. It's only changing it for headers in the response.